src/semgrep_agent/meta.py

```diff
@@ -123,7 +123,10 @@ def commit_sha(self) -> Optional[str]:
     @cachedproperty
     def base_commit_ref(self) -> Optional[str]:
         if self.event_name == "pull_request":
-            return self.glom_event(T["pull_request"]["base"]["sha"])  # type: ignore
+            pr_base = self.glom_event(T["pull_request"]["base"]["sha"])
+            # The pull request "base" that GitHub sends us is not necessarily the merge base,
+            # so we need to get the merge-base from Git
+            return git("merge-base", pr_base, "HEAD").stdout.decode().strip()
```

Comment:
Is this specific to Github or will Gitlab base_commit_ref face the same issue?

Reply:
It looks like GitLab already does something with `merge-base`, although I don't understand what `fetch-head` is as a ref, so I didn't monkey with it.

Reply:
(see line 209)